provisioner/src/lib.rs

```diff
@@ -478,6 +478,18 @@ impl Provisioner for ShuttleProvisioner {
         if !ProjectName::is_valid(&request.project_name) {
             return Err(Status::invalid_argument("invalid project name"));
         }
+
+        // Check project ownership.
+        if !claim
+            .owns_project(&self.gateway_client, &request.project_name)
+            .await
+            .map_err(|_| Status::internal("can not verify project ownership"))?
+        {
+            return Err(Status::unauthenticated(
+                "the request lacks the authorizations",
+            ));
+        }
```

Comment:
Should we emit an error event here if this is false, since if this happens, it's likely an attempt at something malicious? It seems like `Status` implements Error, so we could do `error = &status as &dyn std::error::Error`.

Reply:
good idea, done!